class.jetpack-gutenberg.php

```diff
@@ -58,12 +60,76 @@ public static function load_blocks() {
 			return;
 		}
 
-		foreach ( self::$blocks as $type => $args ) {
-			register_block_type(
-				'jetpack/' . $type,
-				$args
+		/**
+		 * Filter the list of blocks that are available though jetpack.
+		 *
+		 * This filter is populated by Jetpack_Gutenberg::jetpack_set_available_blocks
+		 *
+		 * @since 6.8.0
+		 *
+		 * @param array
+		 */
+		self::$blocks_index = apply_filters( 'jetpack_set_available_blocks', array() );
+
+		foreach ( self::$jetpack_blocks as $type => $args ) {
+			if ( isset( $args['availability']['available'] ) && $args['availability']['available'] && in_array( $type, self::$blocks_index ) ) {
+				register_block_type( 'jetpack/' . $type, $args['args'] );
+			}
+		}
+	}
+
+	public static function preset_exists( $preset ) {
+		return file_exists( JETPACK__PLUGIN_DIR . '/_inc/blocks/' . $preset . '.json' );
+	}
+
+	public static function get_preset( $preset ) {
+		return json_decode( file_get_contents(  JETPACK__PLUGIN_DIR . '/_inc/blocks/' . $preset . '.json' ) );
+	}
+
+	public static function jetpack_set_available_blocks( $blocks ) {
+		$preset_blocks_manifest =  self::preset_exists( 'block-manifest' ) ? self::get_preset( 'block-manifest' ) : (object) array( 'blocks' => $blocks );
+		$preset_blocks = isset( $preset_blocks_manifest->blocks ) ? (array) $preset_blocks_manifest->blocks : array() ;
+		if ( Jetpack_Constants::is_true( 'JETPACK_BETA_BLOCKS' ) ) {
+			$beta_blocks = isset( $preset_blocks_manifest->betaBlocks ) ? (array) $preset_blocks_manifest->betaBlocks : array();
+			return array_merge( $preset_blocks, $beta_blocks );
+		}
+
+		return $preset_blocks;
+	}
+
+	public static function get_block_availability() {
+
+		if ( ! self::should_load_blocks() ) {
+			return array();
+		}
+
+		$blocks_availability = array(); // default
+
+		foreach ( self::$jetpack_blocks as $type => $args ) {
+			if ( ! in_array( $type,  self::$blocks_index ) ) {
+				// Jetpack shouldn't expose blocks that are not in the manifest.
+				continue;
+			}
+			$availability = $args['availability'];
+			$available = array(
+				'available' => ( isset( $availability['available'] ) ? (bool) $availability['available'] : true ),
 			);
+			$unavailability_reason = array();
+			if ( ! $available['available'] ) {
+				$unavailability_reason = array(
+					'unavailable_reason' => ( isset( $availability['unavailable_reason'] ) ? $availability['unavailable_reason'] : 'unknown' )
+				);
+			}
+			$blocks_availability[ $type ] = array_merge( $available, $unavailability_reason );
 		}
+
+		foreach ( self::$blocks_index as $block ) {
+			if ( ! isset( $blocks_availability[ $block ] ) ) {
+				$blocks_availability[ $block ] = array( 'available' => false, 'unavailable_reason' => 'missing_module' );
+			}
+		}
+
+		return $blocks_availability;
```

Comment:
All blocks appear as unavailable on my test site currently running a freshly built set of blocks and with `JETPACK_BETA_BLOCKS` set to true.

```
var Jetpack_Editor_Initial_State = {"available_blocks":{"markdown":{"available":false,"unavailable_reason":"missing_module"},"publicize":{"available":false,"unavailable_reason":"missing_module"},"simple-payments":{"available":false,"unavailable_reason":"missing_module"},"related-posts":{"available":false,"unavailable_reason":"missing_module"},"vr":{"available":false,"unavailable_reason":"missing_module"}},"jetpack":{"is_active":true}};
```

The Markdown module is not active on this site so I expect it to be unavailable, Simple Payments should not be available either since this is a free site, but the others should appear.

Yet I can see the blocks just fine in the editor:
<img width="393" alt="screenshot 2018-11-07 at 18 16 04" src="https://user-images.githubusercontent.com/426388/48148115-48064e00-e2b9-11e8-9559-1f87301e1406.png">

Here is the content of my manifest:

```
{
  "blocks": [
    "markdown",
    "publicize",
    "simple-payments"
  ],
  "betaBlocks": [
    "related-posts",
    "vr"
  ]
}
```

Reply:
This is all expected! We need to do some further work on the SDK to make this work in a reasonable way.

This PR doesn't really do anything, just exposes the data, and that data looks correct to me. As long as there are no regressions, i think we are good. Markdown is the only block that currently respects availability data.